packages/html-reporter/src/testResultView.tsx

```diff
@@ -26,30 +26,65 @@ import { AttachmentLink } from './links';
 import { statusIcon } from './statusIcon';
 import './testResultView.css';
 
-const imageDiffNames = ['diff', 'expected', 'actual'];
+type DiffTab = {
+  id: string,
+  title: string,
+  attachment: TestAttachment,
+};
 
 export const TestResultView: React.FC<{
   test: TestCase,
   result: TestResult,
 }> = ({ result }) => {
 
-  const { screenshots, videos, traces, otherAttachments, attachmentsMap } = React.useMemo(() => {
-    const attachmentsMap = new Map<string, TestAttachment>();
+  const { screenshots, videos, traces, otherAttachments, diffs } = React.useMemo(() => {
     const attachments = result?.attachments || [];
-    const otherAttachments = new Set<TestAttachment>(attachments);
-    const screenshots = attachments.filter(a => a.contentType.startsWith('image/') && !imageDiffNames.includes(a.name));
+    const screenshots = new Set(attachments.filter(a => a.contentType.startsWith('image/')));
     const videos = attachments.filter(a => a.name === 'video');
     const traces = attachments.filter(a => a.name === 'trace');
-    for (const a of attachments)
-      attachmentsMap.set(a.name, a);
+
+    const otherAttachments = new Set<TestAttachment>(attachments);
     [...screenshots, ...videos, ...traces].forEach(a => otherAttachments.delete(a));
-    return { attachmentsMap, screenshots, videos, otherAttachments, traces };
+
+    const snapshotNameToDiffTabs = new Map<string, DiffTab[]>();
+    let tabId = 0;
+    for (const attachment of attachments) {
+      const match = attachment.name.match(/^(.*)-(actual|diff|expected|previous)(\.[^.]+)?$/);
```

Comment:
Hmm.... I thought that if we find `foo-diff.png`, we would also pick up `foo-grayscale.png`, `foo-overlayed.png`, `foo-actual.png` and similar. wdyt?

Reply:
- if there's one from diff, actual or expected, then pick everybody.
- extract this all thing to `classifyAttachments`